src/Makefile.am

```diff
@@ -341,8 +341,8 @@ dicts/emoji-%.dict: emoji-parser
 	    echo "Skip $$plus_comment $@"; \
 	fi;
 
-ibusemojigen.h:
-	$(MAKE) $(AM_MAKEFLAGS) dicts/emoji-en.dict
+ibusemojigen.h: dicts/emoji-en.dict
```

Comment:
If I run `make -C src -j2`, `emoji-parser` is sometimes built failed due to no libibus.so.
So probably I think `ibusemojigen.h: $(libibus) dicts/emoji-en.dict` is needed here and also the similar change for ibusunicodegen.h.

Reply:
Can you show me a log?

It's not clear to me how `emoji-parser` could fail to build without libibus.so because we have

````
emoji_parser_LDADD =            \
    $(libibus)                  \
    $(GLIB2_LIBS)               \
    $(GOBJECT2_LIBS)            \
    $(NULL)
````

With my patches, we have

- `ibusemojigen.h` → `dicts/emoji-en.dict`
- `dicts/emoji-%.dict` → `emoji-parser`
- `emoji-parser` → `$(libibus)`

So you must be referring to some other intermediate step that is failing.

Reply:
I used this script to try to reproduce some failure:

````bash
#!/bin/bash

for x in {1..10}; do
        git clean -dfx
        git checkout -f

        VALAC=/usr/bin/valac-0.56 VAPIGEN=/usr/bin/vapigen-0.56 ./autogen.sh --with-ucd-dir=/usr/share/unicode-data --disable-vala

        make -j8 |& tee -a ../ibus-build-${x}.log
done
````

and I never saw a failure.

Reply:
I can reproduce the issue with 90% in ja_JP.UTF-8 locale in Fedora 38 but cannot with V=1 make argument or LANG=C.
```
# ./autogen.sh
# env LC_MESSAGES=C make -C src -j2
make: Entering directory '/root/rpmbuild/BUILD/ibus-main/src'
  GEN      ibusmarshalers.h
  GEN      ibusenumtypes.c
INFO: Reading ./ibusmarshalers.list...
  GEN      ibusenumtypes.h
  GEN      ibus.gresources.xml
  CC       gen_internal_compose_table-gencomposetable.o
  CC       gen_internal_compose_table-ibuscomposetable.o
  CC       gen_internal_compose_table-ibuserror.o
  CC       gen_internal_compose_table-ibuskeynames.o
  CC       gen_internal_compose_table-ibuskeyuni.o
  GEN      .gitignore
  GEN      ibusmarshalers.c
INFO: Reading ./ibusmarshalers.list...
  CCLD     gen-internal-compose-table
  GEN      compose/sequences-little-endian
  GEN      ibusresources.c
  GEN      ibusresources.h
make  all-recursive
make[1]: Entering directory '/root/rpmbuild/BUILD/ibus-main/src'
Making all in .
make[2]: Entering directory '/root/rpmbuild/BUILD/ibus-main/src'
  CC       emoji_parser-emoji-parser.o
  CC       libibus_1_0_la-ibuscomposetable.lo
  CC       libibus_1_0_la-ibusenumtypes.lo
  CC       libibus_1_0_la-ibusmarshalers.lo
  CC       libibus_1_0_la-ibusresources.lo
  CC       libibus_1_0_la-ibusaccelgroup.lo
  CC       libibus_1_0_la-ibusattribute.lo
  CC       libibus_1_0_la-ibusattrlist.lo
  CC       libibus_1_0_la-ibusbus.lo
  CC       libibus_1_0_la-ibuscomponent.lo
  CC       libibus_1_0_la-ibusconfig.lo
  CC       libibus_1_0_la-ibusconfigservice.lo
  CC       libibus_1_0_la-ibusemoji.lo
  CC       libibus_1_0_la-ibusengine.lo
  CC       libibus_1_0_la-ibusenginedesc.lo
  CC       libibus_1_0_la-ibusenginesimple.lo
  CC       libibus_1_0_la-ibuserror.lo
  CC       libibus_1_0_la-ibusfactory.lo
  CC       libibus_1_0_la-ibushotkey.lo
  CC       libibus_1_0_la-ibusinputcontext.lo
  CC       libibus_1_0_la-ibuskeymap.lo
  CC       libibus_1_0_la-ibuskeynames.lo
  CC       libibus_1_0_la-ibuskeyuni.lo
  CC       libibus_1_0_la-ibuslookuptable.lo
  CC       libibus_1_0_la-ibusobject.lo
  CC       libibus_1_0_la-ibusobservedpath.lo
  CC       libibus_1_0_la-ibuspanelservice.lo
  CC       libibus_1_0_la-ibusproperty.lo
  CC       libibus_1_0_la-ibusproplist.lo
  CC       libibus_1_0_la-ibusproxy.lo
  CC       libibus_1_0_la-ibusregistry.lo
  CC       libibus_1_0_la-ibusserializable.lo
  CC       libibus_1_0_la-ibusservice.lo
  CC       libibus_1_0_la-ibusshare.lo
  CC       libibus_1_0_la-ibustext.lo
  CC       libibus_1_0_la-ibusunicode.lo
  CC       libibus_1_0_la-ibusutil.lo
  CC       libibus_1_0_la-ibusxevent.lo
  CC       libibus_1_0_la-ibusxml.lo
  CC       unicode_parser-unicode-parser.o
make  dicts/emoji-en.dict
make[3]: Entering directory '/root/rpmbuild/BUILD/ibus-main/src'
  CCLD     libibus-1.0.la
make  dicts/unicode-blocks.dict
make[3]: Entering directory '/root/rpmbuild/BUILD/ibus-main/src'
  CCLD     libibus-1.0.la
  CCLD     emoji-parser
/usr/bin/ld: cannot find ./.libs/libibus-1.0.so: No such file or directory
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:1250: emoji-parser] Error 1
make[3]: Leaving directory '/root/rpmbuild/BUILD/ibus-main/src'
make[2]: *** [Makefile:3600: ibusemojigen.h] Error 2
make[2]: *** Waiting for unfinished jobs....
  CCLD     unicode-parser
Generated dicts/unicode-blocks.dict
make[3]: Leaving directory '/root/rpmbuild/BUILD/ibus-main/src'
make[2]: Leaving directory '/root/rpmbuild/BUILD/ibus-main/src'
make[1]: *** [Makefile:2961: all-recursive] Error 1
make[1]: Leaving directory '/root/rpmbuild/BUILD/ibus-main/src'
make: *** [Makefile:1165: all] Error 2
make: Leaving directory '/root/rpmbuild/BUILD/ibus-main/src'

```

Reply:
````
make  dicts/emoji-en.dict
make[3]: Entering directory '/root/rpmbuild/BUILD/ibus-main/src'
  CCLD     libibus-1.0.la
make  dicts/unicode-blocks.dict
make[3]: Entering directory '/root/rpmbuild/BUILD/ibus-main/src'
  CCLD     libibus-1.0.la
  CCLD     emoji-parser
/usr/bin/ld: cannot find ./.libs/libibus-1.0.so: No such file or directory
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:1250: emoji-parser] Error 1
make[3]: Leaving directory '/root/rpmbuild/BUILD/ibus-main/src'
make[2]: *** [Makefile:3600: ibusemojigen.h] Error 2
make[2]: *** Waiting for unfinished jobs....
````

You don't appear to have the patches from this PR applied. Notice `make  dicts/unicode-blocks.dict`. `make` is not executed to build that file after commit `build: Build ibusunicodegen.h without separate make invocation`.

The problem that you mention appears to be exactly what this PR is fixing, so I am very confused about what you are testing.

Reply:
Interesting.
I didn't know `emoji_parser_LDADD = $(libibus) ...` can let `emoji-parser` depend `$(libibus)` and the dependency could be broken by another `$(MAKE)`.
Seems I also didn't know `dicts/unicode-blocks.dict ibusunicodegen.h:` let `make ibusunicodegen.h` try to make dicts/unicode-blocks.dict even if there is no line how to make ibusunicodegen.h.
So I didn't think my issue is relative with your patch but tested in the main repo.

Probably I need to comment on the git description about this more clearly.

Reply:
I am absolutely baffled. What did you think I was fixing? Did you even read the commits?

At this point, I do not think it is a productive for me to spend more time on this since you do not seem to be sufficiently engaged. I suggest merging the commits as they are without modification.

Reply:
I mean your fix is good and I didn't know the logic. Thank you.

But probably  I need to comment on the git description about this more clearly since your explanation was not clear for me.